### PR TITLE
Deprecate WasbPrefixSensorAsync

### PR DESCRIPTION
**Pull request by pankajkoti**
Deprecate WasbPrefixSensorAsync and proxy it to its Airflow 
OSS provider's counterpart

closes: #1412

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -128,6 +128,7 @@ all =
     apache-airflow-providers-http
     apache-airflow-providers-snowflake
     apache-airflow-providers-sftp
+    # TODO: Increment microsoft-azure version as per the upcoming release for deprecating WasbPrefixSensorAsync
```
Comment (pankajkoti) [inline setup.cfg:131]: This needs to be taken care of by updating the corresponding upcoming release version that includes the Airflow OSS provider PR https://github.com/apache/airflow/pull/36806 is released

File: astronomer/providers/microsoft/azure/sensors/wasb.py [hunk 2]
```diff
@@ -48,31 +43,28 @@ def __init__(
...
+    Use :class: `~airflow.providers.microsoft.azure.sensors.wasb.WasbPrefixSensor` instead
+    and set `deferrable` param to `True` instead.
     """
 
     def __init__(
         self,
-        *,
-        container_name: str,
-        prefix: str,
+        *args,
```
Comment (pankajkoti) [inline astronomer/providers/microsoft/azure/sensors/wasb.py:53]:
```suggestion
        *args: Any,
```

**Comment by pankajkoti** [comment]
The implementation work is complete here, but it's waiting for OSS provider release

**Comment by pankajkoti** [comment]
Verified with RC 9.0.0 that the DAG enters deferral and then succeeds

Screenshot 2024-01-31 at 1 47 19 PM

Screenshot 2024-01-31 at 1 48 39 PM

File: setup.cfg [hunk 3]
```diff
@@ -67,7 +67,7 @@ google =
 http =
     apache-airflow-providers-http>=4.9.0
 microsoft.azure =
-    apache-airflow-providers-microsoft-azure>=8.5.1
+    apache-airflow-providers-microsoft-azure>=9.0.0rc1
```
Comment (pankajkoti) [inline setup.cfg:70]:
```suggestion
    apache-airflow-providers-microsoft-azure>=9.0.0
```
Apply once RC is released

File: setup.cfg [hunk 4]
```diff
@@ -127,7 +127,8 @@ all =
     apache-airflow-providers-http>=4.9.0
     apache-airflow-providers-snowflake>=5.3.0
     apache-airflow-providers-sftp
-    apache-airflow-providers-microsoft-azure>=8.5.1
+    # TODO: Increment microsoft-azure version as per the upcoming release for deprecating WasbPrefixSensorAsync
+    apache-airflow-providers-microsoft-azure>=9.0.0rc1
```
Comment (pankajkoti) [inline setup.cfg:131]:
```suggestion
    apache-airflow-providers-microsoft-azure>=9.0.0
```
Apply once RC is released

File: astronomer/providers/microsoft/azure/sensors/wasb.py [hunk 5]
```diff
@@ -82,38 +74,8 @@ def __init__(
...
-        Relies on trigger to throw an exception, otherwise it assumes execution was
-        successful.
-        """
-        if event:
-            if event["status"] == "error":
-                raise_error_or_skip_exception(self.soft_fail, event["message"])
-            self.log.info(event["message"])
+        if kwargs.get("check_options") is None:
+            kwargs["check_options"] = {}
+        kwargs["check_options"]["include"] = include
```
Comment (Lee-W) [inline astronomer/providers/microsoft/azure/sensors/wasb.py:79]: If "check_otpion" is None, wouldn't it causes `KeyError`?

Comment (pankajkoti) [reply astronomer/providers/microsoft/azure/sensors/wasb.py:79]: get does not cause a KeyError

Comment (pankajkoti) [reply astronomer/providers/microsoft/azure/sensors/wasb.py:79]: For other nested keys, we initialise it to empty dictionary just above that line when check_options is None

**Pull request merged**